### DISCUSSION TITLE
Inference Issues with Side View

### DISCUSSION THREAD

**Discussion by CayMurray** [talmolab/sleap#2490] [Help!]
Hello,

I am working on a project where mice in a small open field are recorded from multiple camera angles (1 aerial view and 2 side views). Using the top-down model, I had no issues getting instances involving the head, ears, and tail base using the aerial view. When I move to either side view, I cannot get the model to predict a single instance on any frame. I have played around with the receptive field sizes for both the centroid and centered_instance models, but that didn't give me anything. The training loss curves look reasonable, and when I run training using the UI, the preview I get of predicted instances for the validation frames look accurate. 

I have attached a sample of the types of hyperparameters I have used. I understand that side views are inherently more difficult to score, so perhaps I just need to add more frames (I am currently working with ~200 frames total with a validation fraction of 0.1) or try a different network architecture. I am also unsure if the default augmentation settings are appropriate given that I am using side views. 

<img width="1998" height="1129" alt="Screenshot from 2025-12-08 17-47-34" src="https://github.com/user-attachments/assets/17c180dc-8776-4b0d-b002-604fefa29eec" />
 
<img width="1851" height="1176" alt="Screenshot from 2025-12-08 17-35-00" src="https://github.com/user-attachments/assets/fb1aae8c-e358-4ce4-b98d-f9610516edc7" />

<img width="1851" height="1176" alt="Screenshot from 2025-12-08 17-35-13" src="https://github.com/user-attachments/assets/ef96e6be-763e-4df2-9c69-0ee7a6e12090" />

<img width="1851" height="1176" alt="Screenshot from 2025-12-08 17-35-28" src="https://github.com/user-attachments/assets/c0b37981-46db-44a4-92b5-0ca8b783c3d9" />

Cayden M.

**Comment by eberrigan** [comment]
Hello @CayMurray,

Is it possible the mouse is changing scale a lot from the side views? If it is, you can try using bottom-up instead which is not as sensitive to changes in scale.
It's a little difficult for me to see the receptive field in your picture, but I suggest making it approximately the size of your animal. You can change the input scaling to 0.5 to get the whole animal within the receptive field, and save on computation. You can also try increasing the max stride.
https://docs.sleap.ai/latest/learnings/configuring-models/#hyperparameters

I noticed you have a lot of your nodes visibility turned off. If that is consistent, the model could learn to predict occlusion too frequently. You can try turning those nodes on when it is possible to infer the locations of the nodes. The model will learn to do that too.

Please let us know if you have any questions.

Thanks!

Elizabeth
